src/Symfony/Component/Routing/Tests/Generator/Dumper/CompiledUrlGeneratorDumperTest.php

```diff
@@ -231,4 +231,29 @@ public function testDumpWithSchemeRequirement()
         $this->assertEquals('https://localhost/app.php/testing', $absoluteUrl);
         $this->assertEquals('/app.php/testing', $relativeUrl);
     }
+
+    public function testDumpWithLocalizedRoutesPreserveTheGoodLocaleInTheUrl()
+    {
+        $this->routeCollection->add('foo.en', (new Route('/{_locale}/foo'))->setDefault('_locale', 'en')->setDefault('_canonical_route', 'foo'));
+        $this->routeCollection->add('foo.fr', (new Route('/{_locale}/foo'))->setDefault('_locale', 'fr')->setDefault('_canonical_route', 'foo'));
+        $this->routeCollection->add('fun.en', (new Route('/fun'))->setDefault('_locale', 'en')->setDefault('_canonical_route', 'fun'));
+        $this->routeCollection->add('fun.fr', (new Route('/amusant'))->setDefault('_locale', 'fr')->setDefault('_canonical_route', 'fun'));
+
+        file_put_contents($this->testTmpFilepath, $this->generatorDumper->dump());
+
+        $requestContext = new RequestContext();
+        $requestContext->setParameter('_locale', 'fr');
+
+        $compiledUrlGenerator = new CompiledUrlGenerator(require $this->testTmpFilepath, $requestContext, null, null);
+
+        $this->assertSame('/fr/foo', $compiledUrlGenerator->generate('foo'));
+        $this->assertSame('/en/foo', $compiledUrlGenerator->generate('foo.en'));
+        $this->assertSame('/en/foo', $compiledUrlGenerator->generate('foo', ['_locale' => 'en']));
+        $this->assertSame('/en/foo', $compiledUrlGenerator->generate('foo.fr', ['_locale' => 'en']));
+
+        $this->assertSame('/amusant', $compiledUrlGenerator->generate('fun'));
+        $this->assertSame('/fun', $compiledUrlGenerator->generate('fun.en'));
+        $this->assertSame('/fun', $compiledUrlGenerator->generate('fun', ['_locale' => 'en']));
+        $this->assertSame('/amusant', $compiledUrlGenerator->generate('fun.fr', ['_locale' => 'en']));
```

Comment:
Actually, what do you think of deprecating entirely passing a _locale parameter on a "direct" i18n route?

Reply:
the current behavior is the simpler one, ie the one that requires the less "ifs"/code in the code base, isn't it? then it should stay to me.